src/OpenLoco/src/Audio/Audio.cpp

```diff
@@ -1066,18 +1073,36 @@ namespace OpenLoco::Audio
         {
```

Comment:
This is all starting to get quite complex. Why don't we just ignore the exclude track to begin with.

1. Create the playlist for the different playlist types.
2. If empty add all tracks

4. Find exclude track on the result playlist
5. If exclude track and > 1 tracks remove exclude track from playlist
6. Randomly select from result.

Reply:
I completely agree. I decided to make this as a different pull request though: #2699